### PR TITLE
[BUGFIX] allow translation of children in already translated container

### PR DESCRIPTION
**Pull request by achimfritz**
Resolves: #85

### CODE REVIEW

File: ext_localconf.php [hunk 1]
```diff
@@ -32,6 +32,11 @@
         );
     }
 
+    // LcoalizstionController Xclass
```
Comment (ervaude) [inline ext_localconf.php:35]: LocalizationController

File: Classes/Xclasses/LocalizationController.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+        if (empty($uids)) {
+            return $recordsPerColPos;
+        }
+        $containerUids = $this->getContainerUids($uids);
+        if (empty($containerUids)) {
+            return $recordsPerColPos;
+        }
+        $containerChildren = $this->getContainerChildren($uids);
+        if (empty($containerChildren)) {
```
Comment (ervaude) [inline Classes/Xclasses/LocalizationController.php:73]: $containerChildren gave me a hard time to understand. Would name that $contentInContainers

File: Classes/Xclasses/LocalizationController.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        $this->containerRegistry = $containerRegistry ?? GeneralUtility::makeInstance(Registry::class);
+    }
+
+    public function getRecordLocalizeSummary(ServerRequestInterface $request): ResponseInterface
+    {
+        $response = parent::getRecordLocalizeSummary($request);
+        $payload = json_decode($response->getBody()->getContents(), true);
+        $payload = $this->rebuildPayload($payload);
+        return new JsonResponse($payload);
+    }
```
Comment (ervaude) [inline Classes/Xclasses/LocalizationController.php:42]: Maybe keep everything but this method out if the XClass so it does not seem so bloated.

File: Classes/Xclasses/LocalizationController.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    protected function rebuildPayload(array $payload): array
+    {
+        $recordsPerColPos = $payload['records'];
+        $columns = $payload['columns'];
+        $columns = $this->rebuildColumns($columns);
+        $filteredRecordsPerCol = $this->filterRecords($recordsPerColPos);
+        return [
```
Comment (ervaude) [inline Classes/Xclasses/LocalizationController.php:50]: $columns and $filteredRecordsPerCol are unnecessary variable assignments

File: Tests/Unit/Xclasses/LocalizationControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+use TYPO3\TestingFramework\Core\Unit\UnitTestCase;
+
+class LocalizationControllerTest extends UnitTestCase
+{
+    protected $resetSingletonInstances = true;
+
+    /**
+     * @test
+     */
+    public function filterRecordsRemovesContainerChildren(): void
```
Comment (ervaude) [inline Tests/Unit/Xclasses/LocalizationControllerTest.php:24]: filterRecordsRemovesContainerChildrenIfParentContainerIsTranslatedAsWell

File: Tests/Unit/Xclasses/LocalizationControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        ];
+        $filtered = $localizationController->_call('filterRecords', $recordsToTranslate);
+        self::assertTrue(1 === count($filtered[0]));
+        self::assertTrue(empty($filtered[200]));
+    }
+
+    /**
+     * @test
+     */
+    public function filterRecordsKeepsContainerChildren(): void
```
Comment (ervaude) [inline Tests/Unit/Xclasses/LocalizationControllerTest.php:47]: filterRecordsKeepsContainerChildrenIfParentContainerIsNotTranslated

**Review by ervaude — APPROVED** [review]

File: ext_localconf.php (hunk 1 above)
Comment (achimfritz) [reply ext_localconf.php]: thanks

**Pull request merged**